app/src/main/java/helium314/keyboard/keyboard/internal/keyboard_parser/KeyboardParser.kt

```diff
@@ -88,9 +86,11 @@ class KeyboardParser(private val params: KeyboardParams, private val context: Co
         addNumberRowOrPopupKeys(baseKeys, numberRow)
         if (params.mId.isAlphabetKeyboard)
             addSymbolPopupKeys(baseKeys)
-        if (params.mId.isAlphaOrSymbolKeyboard && params.mId.mNumberRowEnabled)
-            baseKeys.add(0, numberRow
-                .mapTo(mutableListOf()) { it.copy(newLabelFlags = Key.LABEL_FLAGS_DISABLE_HINT_LABEL or defaultLabelFlags) })
+        if (params.mId.isAlphaOrSymbolKeyboard && params.mId.mNumberRowEnabled) {
+            val newLabelFlags =
+                    if (Settings.getInstance().current.mShowNumberRowHints) 0 else Key.LABEL_FLAGS_DISABLE_HINT_LABEL
+            baseKeys.add(0, numberRow.mapTo(mutableListOf()) { it.copy(newLabelFlags = newLabelFlags) })
```

Comment:
```suggestion
            baseKeys.add(0, numberRow.mapTo(mutableListOf()) { it.copy(newLabelFlags = newLabelFlags or defaultLabelFlags) })
```
Do you have a specific reason to omit `defaultLabelFlags` here?